oss_src/unity/python/sframe/_scripts/_pylambda_worker.py

```diff
@@ -0,0 +1,136 @@
+import sys
+import os
+import ctypes
+from ctypes import PyDLL, c_char_p, c_int
+from os.path import split, abspath, join
+from glob import glob
+from itertools import chain
+
+def set_windows_dll_path():
+    """
+    Sets the dll load path so that things are resolved correctly.
+    """
+
+    lib_path = os.path.dirname(os.path.abspath(__file__))
+
+    def errcheck_bool(result, func, args):
+        if not result:
+            last_error = ctypes.get_last_error()
+            if last_error != 0:
+                raise ctypes.WinError(last_error)
+            else:
+                raise OSError
+        return args
+
+    import ctypes.wintypes as wintypes
+
+    # Also need to set the dll loading directory to the main
+    # folder so windows attempts to load all DLLs from this
+    # directory.
+    try:
+        kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+        kernel32.SetDllDirectoryW.errcheck = errcheck_bool
+        kernel32.SetDllDirectoryW.argtypes = (wintypes.LPCWSTR,)
+        kernel32.SetDllDirectoryW(lib_path)
+    except Exception, e:
+        sys.stderr.write("Error setting DLL load orders: %s (things may still work).\n" % str(e))
+        sys.stderr.flush()
+
+if __name__ == "__main__":
+
+    if len(sys.argv) == 1:
+        debug_mode = True
+    else:
+        debug_mode = False
+
+    if debug_mode or os.environ.get("GRAPHLAB_LAMBDA_WORKER_DEBUG_MODE") == "1":
+        _write_out = sys.stderr
+    else:
+        _write_out = sys.stdout
+
+    _write_out_file_name = os.environ.get("GRAPHLAB_LAMBDA_WORKER_LOG_FILE", "")
+    _write_out_file = None
+
+    def _write_log(s, error = False):
+        s = s + "\n"
+
+        if error:
+            sys.stderr.write(s)
+            sys.stderr.flush()
+        else:
+            _write_out.write(s)
+            _write_out.flush()
+
+        if _write_out_file is not None:
+            _write_out_file.write(s)
+            _write_out_file.flush()
+
+    if _write_out_file_name != "":
+
+        # Set this to an absolute location to make things worthwhile
+        _write_out_file_name = abspath(_write_out_file_name)
+        os.environ["GRAPHLAB_LAMBDA_WORKER_LOG_FILE"] = _write_out_file_name
+
+        _write_out_file_name = _write_out_file_name + "-init"
+        _write_log("Logging initialization routines to %s." % _write_out_file_name)
+        try:
+            _write_out_file = open(_write_out_file_name, "w")
+        except Exception, e:
+            _write_log("Error opening '%s' for write: %s" % (_write_out_file_name, repr(e)))
+            _write_out_file = None
+
+    if debug_mode:
+        print "PyLambda script called with no IPC information; entering diagnostic mode."
```

Comment:
Okay, looking at it again, I see that now. It is pretty confusing to have a debug_mode toggled based on the amount of arguments, and then a different debug mode toggled by an environment variable ending in "DEBUG_MODE". Perhaps you could call the one that runs with no IPC a "dry run"?